enterprise/app/code/code_v2.tsx

```diff
@@ -449,32 +478,40 @@ export default class CodeComponentV2 extends React.Component<Props, State> {
     return decor?.options?.after?.attachedData as kythe.proto.DecorationsReply.Reference | undefined;
   }
 
-  getMostSpecificDecoration(range: monaco.Range): kythe.proto.DecorationsReply.Reference | undefined {
-    // Get the decoration that overlaps this range and has the smallest span.
-
+  getKytheRefsForRange(range: monaco.Range): kythe.proto.DecorationsReply.Reference[] | undefined {
+    // This finds the smallest decorations in a given range.
+    // All decorations that share the same smallest span will be returned.
+    // This is necessary because, for example, some definitions will have multiple
+    // /kythe/edge/defines/binding edges, and we want to process them all so we don't
+    // miss references.
     const decorInRange = this.editor?.getDecorationsInRange(range);
     if (!decorInRange) {
       return undefined;
     }
 
-    let minMatch = undefined;
+    let refsInRange = decorInRange.map((decor) => this.decorToReference(decor)).filter((ref) => !!ref);
+
+    let minMatches: kythe.proto.DecorationsReply.Reference[] = [];
     let minMatchLength = Number.POSITIVE_INFINITY;
-    for (const decor of decorInRange) {
-      const data = decor.options.after?.attachedData as kythe.proto.DecorationsReply.Reference;
-      if (!data || !data.span || !data.span.start || !data.span.end) {
+    for (const ref of refsInRange) {
+      if (!ref || !ref.span || !ref.span.start || !ref.span.end) {
```

Comment:
could also do `!(ref?.span?.start && ref?.span?.end)` but this is kinda in the weeds

Reply:
thank you, this is better / I wonder how many brain cycles have been used collectively by humanity on figuring out the best way to do null checks in javascript